x-pack/plugins/security_solution/public/common/components/local_storage/index.tsx

```diff
@@ -37,10 +38,18 @@ export const useLocalStorage = <T,>({
   }, [defaultValue, isInvalidDefault, key, plugin, storage]);
 
   const setValue = useCallback(
-    (value: T) => {
-      storage.set(`${plugin}.${key}`, value);
-
-      _setValue(value);
+    (value: T | ((prev: T) => T)) => {
+      if (typeof value === 'function') {
```

Comment:
Where this is used and how this changes is related to the current PR?

Reply:
It's used in the `useConversationStore` wrapper to store state in localstorage. `_setValue` needed to be extend to support providing the `prev` value as a function param when updating:

https://github.com/elastic/kibana/blob/542f3e5ed5b5df18b4425aa439336e420b4ff542/x-pack/plugins/security_solution/public/assistant/use_conversation_store/index.tsx#L19-L26